common/src/app/common/types/color.cljc

```diff
@@ -213,11 +236,87 @@
                 (= (:ref-file %) library-id))
           all-colors)))
 
+(defn uses-library-color?
+  "Check if the shape uses the given library color."
+  [shape library-id color-id]
+  (let [all-colors (get-all-colors shape)]
+    (some #(and (= (:ref-id %) color-id)
+                (= (:ref-file %) library-id))
+          all-colors)))
+
+(defn- process-shape-colors
+  "Execute an update function on all colors of a shape."
+  [shape func]
```

Comment:
I think a better name can be somethig like `update-fn` instead of `func` that is not very common and not very clear name In my opinion

Reply:
Changed. I prefer the name `process-fn`, because this is not the tipycal update function, that receives an object and returns it modified.

Reply:
:+1: